websites/api-docs-smoke-test/src/type-locations/type-locations.yaml

```diff
@@ -0,0 +1,4 @@
+- type: test__ExternalLinkTestType
```

Comment:
TBD I don't like the data structure design. It's not designed user-first but is backwards derived from the internals of an existing implementation data structure.    Please try to learn building features like this test-first // specification-first.  You write the configuration,  the data file and the usage in the text MDX like it would be ideal and intuitive for the user.  Only then you start implementing. 

Not ideal here: 
 * a single file in a same named folder
 * It's hard to understand why the navigation and top menu yaml is a file in "data" but this one has a custom data structure in a different folder 
 * authors have to know an internal naming convention with two underscores (that's actuallly just an internal hash keying, something like this should never be exposed to a user)
 * "path" is not correct, it contains a hash too. 
 *  example data should also cover links to 
    1. other microsite
    2. completely external URLs
 * since the content is in fact a full href  or URL, consider naming it that way  (or check how authors write such stuff in other places and make it consistent - e.g. in the menu navigation data files)